guides/micronaut-graphql-todo/micronaut-graphql-todo.adoc

```diff
@@ -0,0 +1,398 @@
+
+== What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* JDK 11 or greater installed with `JAVA_HOME` configured appropriately
+* https://www.docker.io/gettingstarted/#h_installation[Docker] installed to run Postgres and to run tests using https://www.testcontainers.org[Testcontainers].
+
+common:completesolution.adoc[]
+
+== Writing the Application
+
+common:cli-or-launch.adoc[]
+
+[source,bash]
+----
+mn create-app example.micronaut.micronautguide \
+    --features=@features@ \
+    --jdk=11 \
+    --build=@build@ \
+    --lang=@lang@
+----
+
+common:build-lang-arguments.adoc[]
+
+common:default-package.adoc[]
+
+If you use Micronaut Launch, select Micronaut Application as application type and add @features-words@ features.
+
+diffLink:[]
+
+common:annotationprocessors.adoc[]
+
+== Persistence layer
+
+=== Schema migration
+
+We will be using Flyway to manage our schema evolution.  We already have the following dependency configured in our build.
+
+dependency:micronaut-flyway[groupId=io.micronaut.flyway]
+
+and Flyway configured in our configuration:
+
+resource:application.yml[tag=flyway]
+
+So we just need to create our initial schema definition:
+
+resource:db/migration/V1__schema.sql[]
+
+<1> We will store authors in a table separate to the ToDos.  They just have a username.
+<2> The table for our ToDos. They have a title, completion status, and a foreign-key reference to the author.
+
+=== Entities
+
+We can then create an Entity class to hold an Author:
+
+source:Author[]
+
+And another to represent a ToDo:
+
+source:ToDo[]
+
+=== Repositories
+
+To allow fetching of these Entities from the database, we can create https://micronaut-projects.github.io/micronaut-data/latest/guide/#dbcRepositories[`JdbcRepositories`] for each.
+
+The simplest of these is the `ToDoRepository` which just requires the default access methods:
+
+source:ToDoRepository[]
+
+<1> We need to specify the dialect, as queries are computed at compile time.
+
+Then we can specify a Repository for our Authors.  We need a few extra finders for this so that we can hook it into GraphQL more easily in the next step:
+
+source:AuthorRepository[]
+
+<1> Again, we need to specify the dialect.
+<2> When creating todos we need to search for an exisiting username.
+<3> When GraphQL loads a ToDo, we will use this to fetch the authors if they are required in the response.
+<4> We need to find an existing author, or else create a new one when creating a ToDo.
+
+== GraphQL
+
+The initial Micronaut application `create-app` step already added the GraphQL dependency into our build for us:
+
+dependency:micronaut-graphql[groupId=io.micronaut.graphql]
+
+So the default GraphQL endpoint `/graphql` is enabled, and you don't need to add any extra configuration.
+
+=== Describe your schema
+
+Create the file `schema.graphqls` in `src/main/resources` directory:
```

Comment:
Remove "in src/main/resources directory" since the full path is displayed above the code block